### ISSUE TITLE
Bigger icons please?

### ISSUE THREAD

**Issue by thany** [transmission-remote-gui/transgui#1177]
It would be helpful to have slightly bigger icons. See screenshot:
<img src="https://user-images.githubusercontent.com/152227/47782117-bbc7ba00-dcff-11e8-8dcb-fc1927b7919f.PNG" alt="screenshot" width="308">

Please note that these icons may appear physically smaller for me than they do for you.
The icons in the menu and tabs suffer the same problem, although the problem isn't as bad because I can read.

This is on Windows 10 x64 on a Dell XPS15 with 4K screen. That's probably the most important bit to know. I'm also on TRGUI 5.15.4 and daemon 2.94.

**Comment by antekgla** [comment]
Menu: View --> Big Toolbar
o Right Click over Toolbar.

See #889 and #1094

**Comment by thany** [comment]
That fixes the toolbar, but:

* Why can this not be done automatically? Screen DPI is not a hidden setting at all. Most applications scale well, without having to set anything manually.
* What if the screen DPI is só high that even "Big Toolbar" is too small? (there are 13" laptops with 4K screens...)
* This doesn't fix the tabs, nor the menu. The items themselves are fine, but the icons are so small they might as well not be there.

**Comment by leonsoft-kras** [comment]
It's simple. Not all programmers have screens with high DPI.

**Comment by leonsoft-kras** [comment]
Suggest how to modify the code correctly. Welcome.

**Comment by thany** [comment]
All UI components appear to scale with the DPI already, and if I'm not mistaken, the OS is doing that automatically (right?). A button with a fixed 16x16 image might be programmed to stick with exactly 16x16 physical pixels. Physical pixels on my screen are absolutely tiny. I guess you could say it might be the equivalent of 6x6 on a "normal" screen.

I suppose that anti-scaling behaviour oughta be removed. To make the icons not look blurry, add bigger icons to accomodate for the automatically growing buttons with HDPI.

Best bet is to use vector icons, but on native win32 application I realize that might be far fetched. So many programs choose to include very high resolution icons, like 64x64 or even more.

**Comment by k0ste** [comment]
Anybody fixed issue with small icons and checkbox in GTK version?

![Selection_001](https://user-images.githubusercontent.com/7759548/57021673-17dfac80-6c57-11e9-81b9-c3e9d6b56db8.png)

**Comment by k0ste** [comment]
Nevermind, transgui is supported build with QT5 and scaled flawless automatically.

I just reduced text scaling to 80%, only one issue I see is a status bar, but is nothing.
![Selection_001](https://user-images.githubusercontent.com/7759548/57023697-e833a300-6c5c-11e9-9fd1-964c4dac03de.png)

**Comment by leonsoft-kras** [comment]
There are almost no enthusiasts for Linux. If you find a solution, let us know.

**Comment by k0ste** [comment]
I can live with this. All other elements scaled perfect.

**Comment by thany** [comment]
There are still a couple of problems to be found in the current version (5.18 as of writing this).
![Capture](https://user-images.githubusercontent.com/152227/67478867-300c1380-f65d-11e9-9c22-1d03773e957c.PNG)
I highlighted them in the screenshot above.

This is on a 15 inch 4K screen, set to 250% zoom. This problem is worse on my other laptop, which has a 13 inch 4K screen set to 300% zoom.

As you can see, I have already anbled "big toolbar", but thats only moving the problem to the future. It appears to display double-sized icons in the toolbar, but screens are likely to get higher and higher resolutions in the future. So it is a workable solution for now, when your zoom level is around-or-about 200%, but in the future 300-400% might become more common.

Idea: is it not possible to use vector icons? Many website do this, so maybe regular programs can do it too? Otherwise include high-res icons (say, 256x256) and scale them down to however big they need to be at the defined zoom level.

**Comment by k0ste** [comment]
Now connection string is fixed in 5.18.
Icons question is still open.
<img width="1914" alt="2019-10-25_09-25" src="https://user-images.githubusercontent.com/7759548/67538556-9babba80-f709-11e9-8ddb-ab4b6ac89e67.png">

**Comment by thany** [comment]
@k0ste So for you the icons are doubled up pixel by pixel. That doesn't happen for me. It's kind of a lazy workaround, but at least it makes the icons somewhat recognizable, especially on very-high DPI screens like mine.
Is there a setting that does that icon doubling for you?

**Comment by k0ste** [comment]
@thany I don't think my icons are doubled. If that - I was not add special options for transgui. Just 300% scaling in Gnome.

**Comment by thany** [comment]
> 
> 
> @thany I don't think my icons are doubled. If that - I was not add special options for transgui. Just 300% scaling in Gnome.

I looked at your screenshot on a normal-DPI monitor and clearly saw your icons doubled in size. And as you can see in my screenshots, my icons are a lot smaller.

So something weird is going on. I assume the application draws icons, not the OS. Still, zooming to 300% in Gnome causes 200% for the icons, but only on Linux. This seems dangerously arbitrary to me. Dangerous in the sense that fixing it for one OS might break it in another.

**Comment by k0ste** [comment]
> So something weird is going on. I assume the application draws icons, not the OS.

Application use frameworks for work with GUI or CLI (like ncurses lib) interfaces. This allow to possibility to run on different OS'es, like transgui can be compiled for Mac, Windows and Linux. You can compile it with QT or GTK support and with this feature application can be faster (because X server/Wayland and ENV Manager will be used less) and looks like stock application.

**Comment by thany** [comment]
Okay. I'm not certain which of Qt or GTK the Windows version uses, but there is a significant difference in icon drawing between it and the Linux version, apparently.

**Comment by aleqcz** [comment]
I have the very same issue as @thany on Windows

**Comment by wilsoncg** [comment]
This issue is quite annoying and does break the core usability of the application, however I have found a workaround until the application code and/or icons are fixed. 

In windows 10 you can select the executable properties and override windows app scaling behaviour, this makes the icons appear regular sized:

<img width="820" height="994" alt="Image" src="https://github.com/user-attachments/assets/143d0f79-1b91-421d-a4d0-c66dc9d7b103" />

<img width="1624" height="538" alt="Image" src="https://github.com/user-attachments/assets/4fe816a3-a893-4b84-a689-ef085b023cf8" />

**Comment by thany** [comment]
A workaround sometimes makes a developer stop caring (developers in general I mean), "because users can fix it themselves". It definitely works though, but it has to be said, the application becomes entirely blurry. And I'm not so sure about the usability of looking at blur 🙂

**Comment by PeterDaveHello** [comment]
Please feel free to help test if #1598 works for you.

**Comment by thany** [comment]
That's a lot of bots chiming in 😯

If I'm going to help test, I'm going to need a way to run the program at that commit, aren't I? And I don't see a release in the past 7 years, so certainly not that one either as a beta or something.

**Comment by k0ste** [comment]
> That's a lot of bots chiming in 😯
> 
> If I'm going to help test, I'm going to need a way to run the program at that commit, aren't I? And I don't see a release in the past 7 years, so certainly not that one either as a beta or something.

Oh, the release of this repository is already a whole meme, even though the maintainer is literally alive. The funniest thing is that @RockstarGames will most likely release GTA VI before new tag of transgui 😀

**Comment by thany** [comment]
Yeah so memes aside, realistically, maybe it's more productive to contribute to [Tremotesf](https://github.com/equeim/tremotesf2) instead 🙂

**Comment by PeterDaveHello** [comment]
Sure. You can either build the PR locally, or use the Windows binary produced by CI if that's more convenient.

The x86_64 CI artifact is available here:
https://github.com/transmission-remote-gui/transgui/actions/runs/33003061123

You should see `transgui-windows-x86_64` under **Artifacts** at the bottom of the page. It contains the build from the PR commit.

If you prefer to build it locally, you can also check out the `fix/issue-1177-high-dpi-icons` branch and build it yourself.

It would be great if you could test it with your high-DPI setup and let us know whether the toolbar, menu, and tab icons now scale properly. A screenshot would also be helpful if anything still looks off.

**Comment by thany** [comment]
@PeterDaveHello Why not make a release then? The last release was 7 years ago, mate...

My suggestion to move to Tremotesf is genuinely useful to people still clinging onto this ancient program. Nothing low-quality about it. And this issue should NOT be closed as long as the fix isn't at least in a release branch, aside from being actually released.

**Comment by PeterDaveHello** [comment]
A new release is a fair request. I haven't cut one yet because a formal release is not just packaging binaries: it means validating the accumulated changes since the last release across Windows, Linux, and macOS, checking that affected features still work both individually and together, and dealing with regressions that may only surface once those changes are exercised as a whole.

For a volunteer-maintained project with limited maintainer time, that is a substantial amount of work—especially while the build and platform coverage themselves are being brought up to date. I do intend to make a release, but I can't honestly give a date.

#1598 is already merged into `master`, where the next release will be cut from. #1177 was closed because the reported defect now has a merged fix, not because a tag exists. Keeping fixed issues open until they eventually ship would turn individual bug reports into placeholders for a future release.

For this bug specifically, the Windows CI build containing the exact change is linked above. The most useful next step is to verify it on a HiDPI setup like the one that originally exposed the problem and report any remaining rendering issues with a screenshot or concrete test result. If that shows the fix is incomplete, reopening the issue would be entirely appropriate.
